src/shared/edit-util.h

```diff
@@ -5,27 +5,33 @@
 
 typedef struct EditFile {
         char *path;
-        char *tmp;
+        char *original_path;
+        char **comment_paths;
+        char *temp;
         unsigned line;
 } EditFile;
 
-#define _EDIT_FILE_FOREACH(s, e, i)                                     \
-        for (typeof(*(e)) *s, *i = (e); (s = i) && i->path; i++)
+typedef struct EditFileContext {
+        EditFile *files;
+        size_t n_files;
+        const char *marker_start;
+        const char *marker_end;
+} EditFileContext;
 
-#define EDIT_FILE_FOREACH(s, e)                                         \
-        _EDIT_FILE_FOREACH(s, e, UNIQ_T(i, UNIQ))
+#define _EDIT_FILES_FOREACH(s, c, i)                                    \
+        for (size_t i = 0, EditFile *s = c.files; i < c.n_files; s += i++)
```

Comment:
I mean something like the following:
```c
for (EditFile *s = c.files; s < c.files + c.n_files; s++)
```
But, after reconsidering about the macro, now I do not think it is necessary to introduce the macro. The pattern is very common (e.g. see systemctl-list-units.c) and not hard to read. (Of course, the original macro conditioned with the existence of `EditFile.path` may not be trivial, but after `EditFileContext` being introduced, the loop is now quite simple.)

Reply:
It still feels tidier so I'd like to preserve it 🤔

Reply:
OK. The opinion about dropping the macro is not strong opinion. Feel free to take any form you'd like.